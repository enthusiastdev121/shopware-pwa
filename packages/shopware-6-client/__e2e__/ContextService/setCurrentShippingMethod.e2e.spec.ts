import {
  setCurrentShippingMethod,
  update,
} from "@shopware-pwa/shopware-6-client";

describe("shopware-6-client - E2E - ContextService - setCurrentShippingMethod", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should set 'Express' shipping method", async () => {
    const result = await setCurrentShippingMethod(
      "abcb43aa3d8a48ba90c70f631ac442c1"
    );
    result.contextToken = "mockedContextToken";
    expect(result).toMatchSnapshot();
  });

  it("should get error response when not existing shipping method", async () => {
    try {
      await setCurrentShippingMethod("qwe");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
