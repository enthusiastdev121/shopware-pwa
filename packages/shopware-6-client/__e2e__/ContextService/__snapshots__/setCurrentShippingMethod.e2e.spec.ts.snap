// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - ContextService - setCurrentShippingMethod should get error response when not existing shipping method 1`] = `[Error: Request failed with status code 400]`;

exports[`shopware-6-client - E2E - ContextService - setCurrentShippingMethod should set 'Express' shipping method 1`] = `
Object {
  "contextToken": "mockedContextToken",
}
`;
